packet.py

```diff
@@ -0,0 +1,55 @@
+class Packet(object):
+    """ A packet class that represents a data packet being sent from
+        a source to a destination
+
+    Args:
+        ns: Instance of the NetworkSimulator class
+        packet_id: Unique id identifying the packet
+        src: The packet's source node
+        dest: The packet's destination node
+        data: Data string being stored in the packet
+
+    Attributes:
+        packet_id: Unique id identifying the packet
+        src: The packet's source node
+        dest: The packet's destination node
+        data: Data string being stored in the packet
+
+    """
+    def __init__(self, ns, packet_id, src=0, dest=0, data=""):
+        self._packet_id = packet_id
+        self._src = src
```

Comment:
src and dest might need to be other other way right? If you send and ack packet. So shouldn't this depend on your packet type?